test/integration/gardenlet/managedseed/managedseed_suite_test.go

```diff
@@ -0,0 +1,239 @@
+	log = logf.Log.WithName(testID)
+
+	features.RegisterFeatureGates()
+
+	By("starting test environment")
+	testEnv = &gardenerenvtest.GardenerTestEnvironment{
+		Environment: &envtest.Environment{},
+		GardenerAPIServer: &gardenerenvtest.GardenerAPIServer{
+			Args: []string{
+				"--disable-admission-plugins=DeletionConfirmation,ResourceReferenceManager,ExtensionValidator,SeedValidator,ShootDNS,ShootQuotaValidator,ShootTolerationRestriction,ManagedSeedShoot,ManagedSeed,ShootManagedSeed,ShootDNS,ShootValidator,SeedValidator",
+			},
+		},
+	}
+
+	var err error
+	restConfig, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(restConfig).NotTo(BeNil())
+
+	DeferCleanup(func() {
+		By("stopping test environment")
+		Expect(testEnv.Stop()).To(Succeed())
+	})
+
+	scheme := kubernetes.GardenScheme
+	Expect(resourcesv1alpha1.AddToScheme(scheme)).To(Succeed())
+
+	By("creating test client")
+	testClient, err = client.New(restConfig, client.Options{Scheme: scheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	testRunID = utils.ComputeSHA256Hex([]byte(uuid.NewUUID()))[:8]
+	log.Info("Using test run ID for test", "testRunID", testRunID)
+
+	By("creating seed")
+	seed = &gardencorev1beta1.Seed{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "seed-",
+			Labels:       map[string]string{testID: testRunID},
+		},
+		Spec: gardencorev1beta1.SeedSpec{
+			Provider: gardencorev1beta1.SeedProvider{
+				Region: "region",
+				Type:   "providerType",
+			},
+			Networks: gardencorev1beta1.SeedNetworks{
+				Pods:     "10.0.0.0/16",
+				Services: "10.1.0.0/16",
+				Nodes:    pointer.String("10.2.0.0/16"),
+			},
+			DNS: gardencorev1beta1.SeedDNS{
+				IngressDomain: pointer.String("someingress.example.com"),
+			},
+		},
+	}
+	Expect(testClient.Create(ctx, seed)).To(Succeed())
+	log.Info("Created Seed for test", "seed", seed.Name)
+
+	DeferCleanup(func() {
+		By("deleting seed")
+		Expect(testClient.Delete(ctx, seed)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("creating garden namespace for test")
+	gardenNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "garden-",
+		},
+	}
+
+	Expect(testClient.Create(ctx, gardenNamespace)).To(Succeed())
+	log.Info("Created Namespace for test", "namespaceName", gardenNamespace.Name)
+
+	DeferCleanup(func() {
+		By("deleting garden namespace")
+		Expect(testClient.Delete(ctx, gardenNamespace)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("setup manager")
+	mgr, err := manager.New(restConfig, manager.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: "0",
+		Namespace:          gardenNamespace.Name,
+		NewCache: cache.BuilderWithOptions(cache.Options{
+			SelectorsByObject: map[client.Object]cache.ObjectSelector{
+				&seedmanagementv1alpha1.ManagedSeed{}: {Label: labels.SelectorFromSet(labels.Set{testID: testRunID})},
+				&gardencorev1beta1.Seed{}:             {Label: labels.SelectorFromSet(labels.Set{testID: testRunID})},
```

Comment:
```suggestion
```
We don't create the Seed explicitly, these Selector shouldn't be added.
After removing this, One of the 2 failing tests pass for me locally.